src/point_one/rtcm/rtcm_framer.h

```diff
@@ -0,0 +1,194 @@
+   *
+   * @param capacity_bytes The maximum framing buffer capacity (in bytes).
+   */
+  explicit RTCMFramer(size_t capacity_bytes)
+      : RTCMFramer(nullptr, capacity_bytes) {}
+
+  /**
+   * @brief Construct a framer instance with a user-specified buffer.
+   *
+   * @post
+   * `buffer` must exist for the lifetime of this instance.
+   *
+   * @param buffer The framing buffer to use. Set to `nullptr` to allocate a
+   *        buffer internally.
+   * @param capacity_bytes The maximum framing buffer capacity (in bytes).
+   */
+  RTCMFramer(void* buffer, size_t capacity_bytes);
+
+  ~RTCMFramer();
+
+  // Don't allow copying or moving to avoid issues with managed buffer_.
+  RTCMFramer(const RTCMFramer&) = delete; // Copy constructor
+  RTCMFramer(RTCMFramer&&) = delete; // Move constructor
+  RTCMFramer& operator=(const RTCMFramer&) = delete; // Copy assignment operator
+  RTCMFramer& operator=(RTCMFramer&&) = delete; // Move assignment operator
+
+  /**
+   * @brief Set the buffer to use for message framing.
+   *
+   * @post
+   * `buffer` must exist for the lifetime of this instance.
+   *
+   * @param buffer The framing buffer to use. Set to `nullptr` to allocate a
+   *        buffer internally.
+   * @param capacity_bytes The maximum framing buffer capacity (in bytes).
+   */
+  void SetBuffer(void* buffer, size_t capacity_bytes);
+
+  /**
+   * @brief Enable/disable warnings for CRC and "message too large" failures.
+   *
+   * This is typically used when the incoming stream has multiple types of
+   * binary content (e.g., interleaved FusionEngine and RTCM messages), and the
+   * RTCM message preamble is expected to appear in the non-RTCM content
+   * occasionally.
+   *
+   * @param enabled If `true`, issue warnings on errors.
+   */
+  void WarnOnError(bool enabled) { warn_on_error_ = enabled; }
+
+  /**
+   * @brief Specify a function to be called when a message is framed.
+   *
+   * @param callback The function to be called with the message header and a
+   *        pointer to the message payload.
+   */
+  void SetMessageCallback(MessageCallback callback) { callback_ = callback; }
+
+  /**
+   * @brief Reset the framer and discard all pending data.
+   */
+  void Reset();
+
+  /**
+   * @brief Process incoming data.
+   *
+   * @param buffer A buffer containing data to be framed.
+   * @param length_bytes The number of bytes to be framed.
+   *
+   * @return The total size of all valid, complete messages, or 0 if no messages
+   *         were completed.
+   */
+  size_t OnData(const uint8_t* buffer, size_t length_bytes);
+
+  /**
+   * @brief Get the number of decoded messages.
+   *
+   * @return The number of RTCM messages successfully decoded.
+   */
+  uint32_t GetNumDecodedMessages() const { return decoded_msg_count_; }
+
+  /**
+   * @brief Get the number of preamble synchronizations that resulted in errors.
+   *
+   * This is not an accurate count of failed messages, but gives an approximate
```

Comment:
"failed messages since the RTCM preamble is not unique and may appear anywhere in the data stream, but"